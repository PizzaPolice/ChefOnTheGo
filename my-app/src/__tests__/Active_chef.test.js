import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import ActiveChef from '../components/active_chef.jsx'


it('ActiveChef component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ActiveChef />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

// This test is shallow rendering our ActiveChef component
describe('ActiveChef Component', () => {
    it('ActiveChef should render without throwing an error', () => {
        expect(shallow(<ActiveChef />).exists(<form className='ActiveChef'></form>)).toBe(true)
      })

      it("should render initial layout", () => {
        // when
        const component = shallow(<ActiveChef />);
        // then
        expect(component.getElements()).toMatchSnapshot();
        })
  
    });