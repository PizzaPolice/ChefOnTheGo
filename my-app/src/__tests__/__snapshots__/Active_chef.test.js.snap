// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveChef Component should render initial layout 1`] = `
Array [
  <div>
    <section
        className="timer"
    >
        <h2>
            You are currently inactive
        </h2>
        <table
            className="timertable"
        >
            <tbody>
                <tr
                    className="info"
                >
                    <td
                        colSpan="3"
                    >
                        You will be active for: 
                    </td>
                </tr>
                <tr>
                    <td
                        id="hours"
                    >
                        HOURS
                    </td>
                    <td
                        id="minutes"
                    >
                        MINUTES
                    </td>
                    <td
                        id="seconds"
                    >
                        SECONDS
                    </td>
                </tr>
                <tr>
                    <td>
                        Hours
                    </td>
                    <td>
                        Minutes
                    </td>
                    <td>
                        Seconds
                    </td>
                </tr>
            </tbody>
        </table>
    </section>
    <form
        action="inactive_chef.html"
        className="stopserve"
        method="post"
    >
        <input
            type="submit"
            value="Stop Serving"
        />
    </form>
</div>,
]
`;
